projects/arlas-toolkit/src/lib/components/search/search.component.ts

```diff
@@ -104,6 +72,44 @@ export class SearchComponent {
     this.filteredSearch = noautocomplete.pipe(mergeWith(autocomplete), mergeWith(nullautocomplete));
   }
 
+  public ngOnInit() {
+    this.keyEvent.pipe(pairwise()).subscribe(l => {
+      if (l[1] === 0 && l[0] !== 0 && this.searchContributor) {
+        this.collaborativeService.removeFilter(this.searchContributor.identifier);
+        this.cdr.detectChanges();
+      }
+    });
+
+    // Retrieve value from the url and future collaborations
+    this.collaborativeService.collaborationBus.pipe(
+      filter(e => e.id === this.searchContributor.identifier || e.id === 'url')
+    ).subscribe(
+      e => {
+        const collaboration = this.collaborativeService.getCollaboration(this.searchContributor.identifier);
+        if (collaboration) {
+          collaboration.filters.forEach((f, collection) => {
+            let initSearchValue = '';
+            if (collection === this.searchContributor.collection) {
+              for (const filter of f) {
+                let searchtxt = filter.q[0][0];
```

Comment:
normally we should not be having any problem, but i suggest that we check that filter.q is an array of length 1 and same for filter.q[0]